app/src/main/java/com/babylon/wallet/android/presentation/settings/accountsecurity/ledgerhardwarewallets/LedgerHardwareWalletsViewModel.kt

```diff
@@ -36,9 +45,50 @@ class LedgerHardwareWalletsViewModel @Inject constructor(
     fun onAddLedgerDeviceClick() {
         viewModelScope.launch {
             if (getProfileUseCase.p2pLinks.first().isEmpty()) {
-                _state.update { it.copy(showContent = LedgerHardwareWalletsUiState.ShowContent.LinkNewConnector) }
+                _state.update { it.copy(showContent = LedgerHardwareWalletsUiState.ShowContent.LinkNewConnector(false)) }
             } else {
-                _state.update { it.copy(showContent = LedgerHardwareWalletsUiState.ShowContent.AddLedger) }
+                if (!state.value.isLinkConnectionEstablished) {
+                    _state.update {
+                        it.copy(
+                            showLinkConnectorPromptState = ShowLinkConnectorPromptState.Show(
+                                ShowLinkConnectorPromptState.Source.UseLedger
+                            )
+                        )
+                    }
+                } else {
+                    _state.update { it.copy(showContent = LedgerHardwareWalletsUiState.ShowContent.AddLedger) }
+                }
+            }
+        }
+    }
+
+    fun dismissConnectorPrompt(linkConnector: Boolean, source: ShowLinkConnectorPromptState.Source) {
+        _state.update {
+            it.copy(
+                showContent = if (linkConnector) {
+                    LedgerHardwareWalletsUiState.ShowContent.LinkNewConnector(
+                        source == ShowLinkConnectorPromptState.Source.AddLedgerDevice
+                    )
+                } else {
+                    it.showContent
+                },
+                showLinkConnectorPromptState = ShowLinkConnectorPromptState.None
+            )
+        }
+    }
+
+    fun disableAddLedgerButtonUntilConnectionIsEstablished() {
+        _state.update {
+            it.copy(
+                showContent = LedgerHardwareWalletsUiState.ShowContent.Details,
+                linkingToConnector = true
+            )
+        }
+        viewModelScope.launch {
+            ledgerMessenger.isConnected.filter { it }.firstOrNull()?.let {
+                _state.update { state ->
+                    state.copy(linkingToConnector = false)
+                }
```

Comment:
this can be moved to the init

Reply:
no it can't. Here I explicitly wait for connection right after I added it